### PR TITLE
RavenDB-22074 - Add an option to specify max items to process in sing…

### PR DESCRIPTION
**Pull request by shiranshalom**
…le expiration and refresh run for v6.0

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22074/Add-an-option-to-specify-max-items-to-process-in-single-expiration-and-refresh-run

### Additional description

PR for v6.0 (https://github.com/ravendb/ravendb/pull/18223) (https://issues.hibernatingrhinos.com/issue/RavenDB-22074/Add-an-option-to-specify-max-items-to-process-in-single-expiration-and-refresh-run#focus=Comments-67-1049764.0-0)

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [x] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [x] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [ ] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [x] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [ ] No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/AbstractBackgroundWorkStorage.cs [hunk 1]
```diff
@@ -61,11 +61,14 @@ private void ThrowWrongDateFormat(Slice lowerId, string expirationDate)
...
-    
-
 
     public Dictionary<Slice, List<(Slice LowerId, string Id)>> GetDocuments(BackgroundWorkParameters options, out Stopwatch duration, CancellationToken cancellationToken)
+    {
+        var totalCount = 0;
+        return GetDocuments(options, ref totalCount, out duration, cancellationToken);
+    }
+
+    public Dictionary<Slice, List<(Slice LowerId, string Id)>> GetDocuments(BackgroundWorkParameters options, ref int totalCount, out Stopwatch duration, CancellationToken cancellationToken)
```
Comment (garayx) [inline src/Raven.Server/Documents/AbstractBackgroundWorkStorage.cs:65]: private and rename to `GetDocumentsInternal`

**Review by garayx — APPROVED** [review]

File: src/Raven.Server/Documents/AbstractBackgroundWorkStorage.cs (hunk 1 above)
Comment (shiranshalom) [reply src/Raven.Server/Documents/AbstractBackgroundWorkStorage.cs:65]: done

**Comment by arekpalinski** [comment]
Why `MaxItemsToProcess` option was not added to the data archival?

**Comment by shiranshalom** [comment]
> Why `MaxItemsToProcess` option was not added to the data archival?

sorry, missed your comment on that, fixed

File: test/SlowTests/Server/Documents/DataArchival/DataArchivalTests.cs [hunk 2]
```diff
@@ -186,6 +185,70 @@ public async Task ThrowsIfUsingWrongArchiveAtDateTimeFormat()
             }
         }
 
+        [RavenTheory(RavenTestCategory.Configuration)]
+        [RavenData(DatabaseMode = RavenDatabaseMode.All)]
+        public async Task ArchiveDocsWithMaxItemsToProcessConfiguredShouldWork(Options options)
+        {
+            using (var store = GetDocumentStore(options))
+            {
+                // Insert documents with refresh before activating the refresh
```
Comment (poissoncorp) [inline test/SlowTests/Server/Documents/DataArchival/DataArchivalTests.cs]: archival* 😉

**Review by poissoncorp — APPROVED** [review]

File: test/SlowTests/Server/Documents/DataArchival/DataArchivalTests.cs (hunk 2 above)
Comment (shiranshalom) [reply test/SlowTests/Server/Documents/DataArchival/DataArchivalTests.cs]: done :)

**Pull request merged**